daemon/mgr/network.go

```diff
@@ -579,12 +580,22 @@ func endpointOptions(n libnetwork.Network, endpoint *types.Endpoint) ([]libnetwo
 		}
 	}
 
+	// generate genric endpoint options
 	genericOption := options.Generic{}
-	createOptions = append(createOptions, libnetwork.EndpointOptionGeneric(genericOption))
-
 	if len(endpoint.GenericParams) > 0 {
-		createOptions = append(createOptions, libnetwork.EndpointOptionGeneric(endpoint.GenericParams))
+		genericOption, _ = utils.MergeMap(genericOption, endpoint.GenericParams)
 	}
+
+	if n.Name() == endpoint.NetworkMode && endpoint.MacAddress != "" {
+		mac, err := net.ParseMAC(endpoint.MacAddress)
+		if err != nil {
+			return nil, err
+		}
+		genericOption, _ = utils.MergeMap(genericOption, options.Generic{netlabel.MacAddress: mac})
```

Comment:
we already have `Merge` funtion in `utils.go`, can we use it ?

Reply:
It can't to merge the map as I want, I don't know and I don't want to knon what's wrong with it. 🐶